lxd/storage/drivers/driver_zfs_volumes.go

```diff
@@ -1936,7 +1936,7 @@ func (d *zfs) getVolumeDiskPathFromDataset(dataset string) (string, error) {
 	zvolUdevLink := filepath.Join("/dev/zvol", dataset)
 
 	// Shortcut for udev.
-	if shared.PathExists(zvolUdevLink) {
+	if shared.PathExists(zvolUdevLink) && shared.IsBlockdevPath(zvolUdevLink) {
```

Comment:
@simondeziel what do you think about using an os.Stat call here for `zvolUdevLink` so that the symlink is followed, and then using that to check for `fs.ErrNotExist` error and then if not passing the return file info to `shared.IsBlockdev()` to avoid doing an os.Lstat and and os.Stat?

Reply:
I added https://github.com/canonical/lxd/pull/15947/commits/a932a8e0659e3b2afbdee88627c2eca311ba228d to that effect. Let me know if I understood your proposed solution wrong.

Reply:
LGTM thanks!